packages/insured-bridge-relayer/src/index.ts

```diff
@@ -51,129 +55,155 @@ export async function run(logger: winston.Logger, l1Web3: Web3): Promise<void> {
-    const filteredL1Whitelist = await pruneWhitelistedL1Tokens(
-      logger,
-      l1Client,
-      l2Client,
-      config.whitelistedRelayL1Tokens
+    const relayers = await Promise.all(
+      config.activatedChainIds.map(async (chainId: number) => {
+        // Construct a web3 instance running on L2.
+        const l2Web3 = getWeb3ByChainId(config.activatedChainIds[0]);
+        const latestL2BlockNumber = await l2Web3.eth.getBlockNumber();
+        const l2StartBlock = Math.max(0, latestL2BlockNumber - config.l2BlockLookback);
+        const fallbackL2Web3s = getRetryWeb3sByChainId(config.activatedChainIds[0]);
+        // Note: This will not construct duplicate Web3 objects for URL's in the Retry config that are the same as the
+        // one used to construct l2Web3.
+        logger.debug({
+          at: "AcrossRelayer#index",
+          message: `Constructed ${fallbackL2Web3s.length} fallback L2 web3 providers`,
+        });
+        const l2Client = new InsuredBridgeL2Client(
+          logger,
+          l2Web3,
+          await l1Client.getL2DepositBoxAddress(config.activatedChainIds[0]),
+          config.activatedChainIds[0],
+          l2StartBlock,
+          null,
+          fallbackL2Web3s
+        );
+
+        // Update the L2 client and filter out tokens that are not whitelisted on the L2 from the whitelisted L1 relay list.
+        const filteredL1Whitelist = await pruneWhitelistedL1Tokens(
+          logger,
+          l1Client,
+          l2Client,
+          config.whitelistedRelayL1Tokens
+        );
+
+        // Construct the profitability calculator based on the filteredL1Whitelist and relayerDiscount.
+        const profitabilityCalculator = new ProfitabilityCalculator(
+          logger,
+          filteredL1Whitelist,
+          l1ChainId,
+          config.relayerDiscount
+        );
+
+        const relayer = new Relayer(
+          logger,
+          gasEstimator,
+          l1Client,
+          l2Client,
+          profitabilityCalculator,
+          filteredL1Whitelist,
+          accounts[0],
+          config.whitelistedChainIds,
+          config.l1DeployData,
+          config.l2DeployData,
+          config.l2BlockLookback
+        );
+
+        const canonicalBridgeAdapter = createBridgeAdapter(logger, l1Web3, l2Web3, config.activatedChainIds[0]);
+        if (config.botModes.l1FinalizerEnabled) await canonicalBridgeAdapter.initialize();
+
+        const crossDomainFinalizer = new CrossDomainFinalizer(
+          logger,
+          gasEstimator,
+          l1Client,
+          l2Client,
+          canonicalBridgeAdapter,
+          accounts[0],
+          config.l2DeployData,
+          config.crossDomainFinalizationThreshold
+        );
+
+        return {
+          relayer,
+          crossDomainFinalizer,
+          l2Client,
+          profitabilityCalculator,
+          filteredL1Whitelist,
+        };
```

Comment:
This is what's inside the array that the promise.all returns. It will be used in the Promise.allSettled loop below.